[Patch] Completed RTM Info

KarimNahas opened the pull request:
PR Content:
- Added missing information collected from `rtm.start` (info is generated when creating a new rtm using `RTMOptionUseStart` option)

Commit:
Added missing information collected from rtm start

kanata2 commented on info.go:
```diff
@@ -432,9 +432,15 @@ type Icons struct {
+	MPIMs    []*Channel   `json:"mpims",omitempty`
+	IMs      []*Channel   `json:"ims",omitempty`
+	Users    []*User      `json:"users,omitempty"`
+	Bots     []*User      `json:"bots,omitempty"`
```
These fields are included only when you use `rtm.start`, so it would be nice if you could add that to comments.

Commit:
Added explanation

Commit:
Patched omitempty lint

KarimNahas commented:
@kanata2 
I've rebased the code and added a description as per your review (I tried to be as explicit as possible). 
Also made sure that all test are passing.
Tell me if any other modifications are needed.

Commit:
Patched rtm sync info (fixed bots type)

Commit:
Bonus: added missing props to ImageBlock (size, height and width)

nlopes commented:
The `rtm.Start` has since been deprecated and shouldn't be used so I'm not adding that. I'll add a PR for the attachments change briefly.
